### ISSUE TITLE
Method suggest on debug build for VC-2013 is extrem slow

### ISSUE THREAD

**Issue by xMRi** [hunspell/hunspell#390]
I just updated from 1.33.
The new code is extrem slow in the debug build of VS-2013.
I already disabled check iterators, and also the CRT heap check is disabled. No change.

I can't see this behavior in the release version. Everything seams OK there. I checked the source for special debug settings but I didn't find any.

**Comment by xMRi** [comment]
I found one location that ist an extreme waste of CPU and memory time.
suggestmgr.cxx lines 1943+1944
The creation of the temporary string is not needed.

change:
```
std::string temp(s1.substr(i, j));
if (t.find(temp) != std::string::npos) {
```
into
`if (t.find(s1.c_str()+i, 0, j) != std::string::npos) {

**Comment by dimztimz** [comment]
I noticed that too some time ago. Can you do profiling?

**Comment by xMRi** [comment]
I did, but the only location I found on the first fast profile sessions was the code location I already mentioned above. The heavy CPU usage and time is inside the ngram analysis.

I didn't dive deeper. Because hunspell is just deep inside our packages I decided to enable full optimizations even for the debug libs. It is just for this code, and I don't believe that I ever will need to debug inside hunspell.
